composer/algorithms/progressive_resizing/progressive_resizing.py

```diff
@@ -197,14 +210,23 @@ def apply(self, event: Event, state: State, logger: Optional[Logger] = None) ->
         # Calculate the current size of the inputs to use
         initial_size = self.initial_scale
         finetune_fraction = self.finetune_fraction
-        scale_frac_elapsed = min([state.get_elapsed_duration().value / (1 - finetune_fraction), 1])
+        dur_elapsed = state.get_elapsed_duration().value
+        if dur_elapsed >= self.delay_fraction:
+            scale_frac_elapsed = min([dur_elapsed / (1 - self.finetune_fraction - self.delay_fraction), 1])
+        else:
+            scale_frac_elapsed = 0.0
 
         # Linearly increase to full size at the start of the fine tuning period
-        scale_factor = initial_size + (1 - initial_size) * scale_frac_elapsed
+        scale_factor = self.initial_scale + (1 - self.initial_scale) * scale_frac_elapsed
+
+        # adjust scale factor so that we make width a multiple of pin_sizes
+        width = input.shape[3]
+        scaled_width_pinned = round(width * scale_factor / self.pin_sizes) * self.pin_sizes
+        scale_factor_pinned = scaled_width_pinned / width
```

Comment:
should this use `max(width, height)` (or max of `input.shape[2:]`) instead of `width`? Or maybe min so that the worst-case dim is a mulitple of pin_sizes? I'm actually not sure what the right semantics are but I would kind of expect the resizing to not designate a particular dimension as special.

Reply:
Width is an inner dimension (NHWC). I made width to be pinned to a multiple of pin_sizes (say 32) and if cuda kernels rely on inner most after reduction dim(C) to be a nice multiple of a power of 2 we get performance benefits. If H == W, H will also be a multiple of pin_sizes otherwise it will be whatever it becomes after multiplying with scale_factor_pinned.